homeassistant/components/fritz/device_tracker.py

```diff
@@ -1,107 +1,195 @@
+        )
+    )
 
-    def __init__(self, config):
-        """Initialize the scanner."""
-        self.last_results = []
-        self.host = config[CONF_HOST]
-        self.username = config[CONF_USERNAME]
-        self.password = config.get(CONF_PASSWORD)
-        self.success_init = True
-
-        # Establish a connection to the FRITZ!Box.
-        try:
-            self.fritz_box = FritzHosts(
-                address=self.host, user=self.username, password=self.password
-            )
-        except (ValueError, TypeError):
-            self.fritz_box = None
-
-        # At this point it is difficult to tell if a connection is established.
-        # So just check for null objects.
-        if self.fritz_box is None or not self.fritz_box.modelname:
-            self.success_init = False
-
-        if self.success_init:
-            _LOGGER.info("Successfully connected to %s", self.fritz_box.modelname)
-            self._update_info()
-        else:
-            _LOGGER.error(
-                "Failed to establish connection to FRITZ!Box with IP: %s", self.host
-            )
+    _LOGGER.warning(
+        "Your Fritz configuration has been imported into the UI, "
+        "please remove it from configuration.yaml. "
+        "Loading Fritz via scanner setup is now deprecated"
+    )
 
-    def scan_devices(self):
-        """Scan for new devices and return a list of found device ids."""
-        self._update_info()
-        active_hosts = []
-        for known_host in self.last_results:
-            if known_host["status"] and known_host.get("mac"):
-                active_hosts.append(known_host["mac"])
-        return active_hosts
-
-    def get_device_name(self, device):
-        """Return the name of the given device or None if is not known."""
-        ret = self.fritz_box.get_specific_host_entry(device).get("NewHostName")
-        if ret == {}:
-            return None
-        return ret
-
-    def get_extra_attributes(self, device):
-        """Return the attributes (ip, mac) of the given device or None if is not known."""
-        ip_device = None
-        try:
-            ip_device = self.fritz_box.get_specific_host_entry(device).get(
-                "NewIPAddress"
-            )
-        except fritzexceptions.FritzLookUpError as fritz_lookup_error:
-            _LOGGER.warning(
-                "Host entry for %s not found: %s", device, fritz_lookup_error
-            )
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for FRITZ!Box component."""
+    _LOGGER.debug("Starting FRITZ!Box device tracker")
+    router = hass.data[DOMAIN][entry.entry_id]
+    tracked = set()
+
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        _async_add_entities(router, async_add_entities, tracked)
+
+    async_dispatcher_connect(hass, router.signal_device_new, update_router)
```

Comment:
We should save the return value and use it to unsubscribe the signal on entry unload.